pkg/pool-weighted/contracts/WeightedPool.sol

```diff
@@ -23,18 +23,80 @@ import "./BaseWeightedPool.sol";
 contract WeightedPool is BaseWeightedPool {
     using FixedPoint for uint256;
 
+    uint256 private constant _MAX_TOKENS = 20;
```

Comment:
This is private but everything else is internal. What is the intended behavior here? Is this to allow for a future contract to derive from this one add more tokens, like your original design of the 8- and 16-token weighted pools?

Reply:
Yes - I do want them to be able to derive from WeightedPool, but it might be confusing to have multiple _MAX_TOKENS defined if this were internal. You'd have _MAX_TOKENS and then _MAX_NEW_POOL_TOKENS.

If they want to keep the same max token count in the new pool, they would just not override _getMaxTokens()